src/menu.c

```diff
@@ -3488,6 +3489,75 @@ static void SARMenuDoDrawObject(
 		    }
 		}
 		while((buf_ptr != NULL) && (lines_drawn < lines_visable));
+
+		    /* Calculate total number of lines */
+		    total_lines = SARMenuMessageBoxTotalLines(
+			mesgbox->message, columns_visable
+		    );
+
+	    /* Draw scroll indicator (not draggable, only a position indicator) */
+        int scroll_box_left, scroll_box_right, scroll_box_bottom, scroll_box_top, indicator_top, indicator_bottom;
+        float indicator_max_height, indicator_height, indicator_top_blank_space;
+	    scroll_box_left = xc + w - DEF_MB_XMARGIN - DEF_SCROLL_CURSOR_WIDTH;
+	    scroll_box_right = xc + w - DEF_MB_XMARGIN - 1;
+	    scroll_box_bottom = height - (y + (h / 2) - (DEF_MB_YMARGIN + DEF_SCROLL_CURSOR_HEIGHT) + 1);
+	    scroll_box_top = height - (yc + DEF_SCROLL_CURSOR_HEIGHT + DEF_MB_YMARGIN) - 1;
+        
+	    glBegin(GL_LINES);
+			/* Draw left border line */
+			glVertex2i(
+                scroll_box_left,
+                scroll_box_top + 1
+            );
```

Comment:
Would it be possible to fix the indenting? It seems like things are indented at 4 different levels.